pybaseball/statcast.py

```diff
@@ -1,175 +1,70 @@
 _MAX_SC_RESULTS = 40000
 
 
-def _chunk_small_request(start_date: date, end_date: date, team: Optional[str], verbose: bool,
-                         max_retries: int = 3) -> Optional[pd.DataFrame]:
-    data = small_request(start_date, end_date, team=team, verbose=verbose)
-
-    # Append to list of dataframes if not empty or failed
-    # (failed requests have one row saying "Error: Query Timeout")
-    if not data.empty:
-        if verbose:
-            print(f"Completed sub-query from {start_date} to {end_date} ({len(data)} results)")
-        return data
-
-    # If it failed, retry up to three times
-    retry_success = False
-
-    # Count failed requests. If > 2, break
-    error_counter = 0
-
-    dataframe_list: List[pd.DataFrame] = []
-    while not retry_success:
-        data = small_request(start_date, end_date, team=team, verbose=verbose)
-
-        if not data.empty:
-            if verbose:
-                print(f"Completed sub-query from {start_date} to {end_date} ({len(data)} results)")
-            return data
-
-        error_counter += 1
-
-        if error_counter < max_retries:
-            continue
-
-        # This request is probably too large. Cut a day off of this request
-        # and make that its own separate request. For each, append to
-        # dataframe list if successful, skip and print error message if failed.
-        if end_date > start_date:
-            tmp_end = end_date - timedelta(days=1)
-            smaller_data_1 = small_request(start_date, tmp_end, team=team, verbose=verbose)
-            smaller_data_2 = small_request(end_date, end_date, team=team, verbose=verbose)
-
-            if smaller_data_1 is not None and not smaller_data_1.empty:
-                dataframe_list.append(smaller_data_1)
-                if verbose:
-                    print(f"Completed sub-query from {start_date} to {tmp_end} ({len(smaller_data_1)} results)")
-                retry_success = True
-            else:
-                print(f"Query unsuccessful for data from {end_date} to {tmp_end}. Skipping these dates.")
-
-            if smaller_data_2 is not None and not smaller_data_2.empty:
-                dataframe_list.append(smaller_data_2)
-                if verbose:
-                    print(f"Completed sub-query from {end_date} to {end_date} ({len(smaller_data_2)} results)")
-                retry_success = True
-            else:
-                print(f"Query unsuccessful for data from {end_date} to {end_date}. Skipping these dates.")
-
-            # Reset counter
-            error_counter = 0
-            break
-
-    if dataframe_list:
-        return pd.concat(dataframe_list, axis=0)
-
-    return None
-
-
+@cache.df_cache(expires=365)
```

Comment:
I don't think you need this here. The function `get_statcast_data_from_csv_url` from `datasources/statcast.py` has a cache on it already, and this calls that function. I would just add the expiration to that existing wrapper.

Reply:
I wanted this here for clarity really, didn't want to run it thru another indirection layer before hitting the cache.